app/models/manageiq/providers/ibm_power_hmc/infra_manager/lpar.rb

```diff
@@ -79,6 +79,14 @@ def process_samples(counters, samples)
     end || {}
   end
 
+  def mapped_disks
+    ext_management_system.disks.joins("INNER JOIN miq_scsi_luns ON miq_scsi_luns.device_name=disks.device_name").joins("INNER JOIN miq_scsi_targets ON miq_scsi_targets.id=miq_scsi_luns.miq_scsi_target_id").joins("INNER JOIN guest_devices ON guest_devices.id=miq_scsi_targets.guest_device_id").where(guest_devices: {ems_ref: ems_ref})
+  end
+
+  def hardware
+    ManageIQ::Providers::IbmPowerHmc::InfraManager::VmHardware.new(super)
```

Comment:
Is this trying to replace the `::Hardware` record that would normally be returned?

Reply:
Yes, this is a wrapper around the hardware method.

In the HMC REST API, details of a volume can be found under the VIOS object providing access to the volume and not on the LPAR object the volume is ultimately attached to. We have preserved this relationship in the HMC provider by storing Disk entries under the Hardware of the VIOS Vm instead of the LPAR Vm, mainly for Targeted Refresh considerations.

In the UI, it would probably be more useful to list volumes under the LPAR Vm. Hence attempt at redirecting UI calls to `@record.hardware.disks` or other disk-related methods to pass through new method `mapped_disks` that returns list of disks actually attached to the VM.